File: joysong-app/app/src/main/java/com/joysong/app/domain/repository/AuthRepository.kt
package com.joysong.app.domain.repository

import com.joysong.app.domain.model.User

interface AuthRepository {
    suspend fun sendVerificationCode(phone: String): Result<Unit>
    suspend fun login(phone: String, password: String): Result<User>
    suspend fun loginWithCode(phone: String, code: String): Result<User>
    suspend fun loginWithGoogle(idToken: String): Result<User>
    suspend fun register(phone: String, code: String, password: String): Result<User>
    suspend fun resetPassword(phone: String, code: String, newPassword: String): Result<Unit>
    suspend fun checkPhoneRegistered(phone: String): Result<Boolean>
    suspend fun getUserProfile(): Result<User>
    suspend fun isLoggedIn(): Boolean
    suspend fun logout()
    suspend fun deleteAccount(): Result<Unit>
}

File: joysong-flutter/lib/features/auth/domain/auth_repository.dart
import 'package:joysong_flutter/features/auth/domain/auth_models.dart';

abstract interface class AuthRepository {
  Future<AuthSession> loginWithPassword({
    required String phone,
    required String password,
  });

  Future<AuthSession> loginWithCode({
    required String phone,
    required String code,
  });

  Future<AuthSession> loginWithGoogle({required String idToken});

  Future<AuthSession> register({
    required String phone,
    required String code,
    required String password,
  });

  Future<void> sendCode({required String phone});

  Future<bool> checkPhoneRegistered({required String phone});

  Future<void> resetPassword({
    required String phone,
    required String code,
    required String newPassword,
  });

  Future<AuthSession> refreshSession();

  Future<void> logout();

  Future<AuthTokens?> readTokens();

  Future<void> activateTokens(AuthTokens tokens);

  Future<void> clearLocalTokens();
}
